modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,198 @@
+        Bson documentEvent = and(eq("category", "eventDocumentCategory"),
+                in("eventId", "documentCreated", "documentModified", "documentMoved", "documentCreatedByCopy",
+                        "documentRestored", "addedToCollection", "documentProxyPublished", "documentLocked",
+                        "documentUnlocked", "documentUntrashed"));
+        Bson sessionRepository = eq("repositoryId", session.getRepositoryName());
+        Bson driveCategory = eq("category", "NuxeoDrive");
+        Bson notRootUnregistered = ne("eventId", "rootUnregistered");
+        Bson inEvents = or(documentEvent, lifeCycleEvent);
+        Bson isGeneral = addRoots(inEvents, activeRoots.getPaths(), collectionSyncRootMemberIds);
+        Bson isDrive = and(driveCategory, notRootUnregistered);
+        Bson idRange = and(gt("_id", lowerBound), lte("_id", upperBound));
+
+        // Build final filter
+        Bson filter = and(sessionRepository, or(isGeneral, isDrive), idRange);
+
+        log.debug("Query on MongoDB-Audit: {}",
+                () -> filter.toBsonDocument(Document.class, auditCollection.getCodecRegistry()).toJson());
+
+        Bson order = orderBy(ascending("repositoryId"), descending("eventDate"));
+        List<LogEntry> entries = auditCollection.find(filter)
+                                                .sort(order)
+                                                .map(MongoDBAuditEntryReader::read)
+                                                .into(new ArrayList<>());
+
+        // Post filter the output to remove (un)registration that are unrelated
+        // to the current user.
+        List<LogEntry> postFilteredEntries = new ArrayList<>();
+        String principalName = session.getPrincipal().getName();
+        for (LogEntry entry : entries) {
+            ExtendedInfo impactedUserInfo = entry.getExtendedInfos().get("impactedUserName");
+            if (impactedUserInfo != null && !principalName.equals(impactedUserInfo.getValue(String.class))) {
+                // ignore event that only impact other users
+                continue;
+            }
+            log.debug("Change detected: {}", entry);
+            postFilteredEntries.add(entry);
+        }
+        return postFilteredEntries;
+    }
+
+    protected Bson addRoots(Bson baseFilter, Set<String> rootPaths, Set<String> collectionSyncRootMemberIds) {
+        if (rootPaths.isEmpty() && collectionSyncRootMemberIds.isEmpty()) {
+            return and(baseFilter);
+        }
+        List<Bson> rootFilters = new ArrayList<>();
+        if (!rootPaths.isEmpty()) {
+            rootFilters.add(regex("docPath", "^" + String.join("|^", rootPaths)));
+        }
+        if (!collectionSyncRootMemberIds.isEmpty()) {
+            rootFilters.add(in("docUUID", collectionSyncRootMemberIds));
+        }
+        if (rootFilters.size() > 1) {
+            return and(baseFilter, or(rootFilters));
+        }
+        return and(baseFilter, rootFilters.get(0));
```

Comment:
This should be clearer:

          List<Bson> rootFilters = new ArrayList<>();
          if (!rootPaths.isEmpty()) {
              rootFilters.add(regex("docPath", "^" + String.join("|^", rootPaths)));
          }
          if (!collectionSyncRootMemberIds.isEmpty()) {
              rootFilters.add(in("docUUID", collectionSyncRootMemberIds));
          }
          if (rootFilters.isEmpty()) {
              return baseFilter;
          } else {
              rootFilters.add(baseFilter);
              return and(rootFilters);
          }

Reply:
Doesn't cut the unit tests because ORs between rootFilters are missing.
If you don't mind making an or(list) with a single element in list (ANDs are only added when necessary in the final query so I guess ORs as well) we can do the following:
```
        if (!rootPaths.isEmpty() && !collectionSyncRootMemberIds.isEmpty()) {
            return and(baseFilter);
        }
        List<Bson> rootFilters = new ArrayList<>();
        if (!rootPaths.isEmpty()) {
            rootFilters.add(regex("docPath", "^" + String.join("|^", rootPaths)));
        }
        if (!collectionSyncRootMemberIds.isEmpty()) {
            rootFilters.add(in("docUUID", collectionSyncRootMemberIds));
        }
        return and(baseFilter, or(rootFilters));
```

Reply:
Hm yeah I misread the original code. But it's ugly to test twice for `rootPaths.isEmpty` and `collectionSyncRootMemberIds.isEmpty()`. Just change my code a bit to do at the end:

          return and(baseFilter, or(rootFilters));